htlcswitch/link.go

```diff
@@ -2387,6 +2401,9 @@ func (l *channelLink) handleUpstreamMsg(msg lnwire.Message) {
 		// Update the mailbox's feerate as well.
 		l.mailBox.SetFeeRate(fee)
 
+	case *lnwire.Stfu:
+		l.handleStfu(msg)
```

Comment:
I think this raises the question of how we should gate messages based off of negotiated features and at what layer. I don't want to duplicate this check at every layer in the stack. Intuitively I think this should be handled by the message bus itself but perhaps that's the wrong way to go about it. I think the two "natural" places for gating feature behavior are either at the top-most level (message bus) or bottom-most level (what you're suggesting right now, I think). The downside about handling it at the bottom most level is that it makes for a lot more code branching.

One of the other solutions here is the make the quiescer an interface and have a noop implementation of that interface that we construct the link with if the feature wasn't negotiated. I mostly don't want to have a bajillion feature flag switches throughout the codebase like we did with Taproot Channels. It's ugly and can be avoided.

Reply:
> One of the other solutions here is the make the quiescer an interface and have a noop implementation of that interface that we construct the link with if the feature wasn't negotiated.

I like this! Can pass that in on the link config level so that it doesn't need to know about it. If the stub always errors out on `handleStfu` then the handling in the link will disconnect anyway, which is the correct response IMO.

> I mostly don't want to have a bajillion feature flag switches

added some for route blinding, feelsbadman